internal/brokers/broker.go

```diff
@@ -90,6 +96,9 @@ func (b Broker) newSession(ctx context.Context, username, lang string) (sessionI
 func (b Broker) GetAuthenticationModes(ctx context.Context, sessionID string, supportedUILayouts []map[string]string) (authenticationModes []map[string]string, err error) {
 	sessionID = b.parseSessionID(sessionID)
 
+	//TODO: One per session or one per broker?
+	b.generateValidators(supportedUILayouts)
```

Comment:
Should this be called every time we call GetAuthenticationModes or just when we don't have the validators (i.e. `b.layoutValidators == nil`)? There is a compelling case for both options, tbh.

Reply:
right, as we will go over this loop multiple times in the future for multiple MFA, we may have different requirements (first one without browser support, then browser is supported for instance), so it’s safest to do it everytime we return (from the PAM perspective) the constraints.